infrastructure/Taskfile.yml

```diff
@@ -897,6 +897,26 @@ tasks:
           fi
       preconditions:
       - *require_site
+      - &require_clean_state {
+          sh: "[
+            current_branch=$(git branch --show-current)
+            if [ 'current_branch' != 'main' ]; then
+              echo 'Can't deploy if not on main branch'
+              exit 1
+            fi
+            if ! git diff-index --quiet HEAD --; then
+              echo 'Can't deploy if there's changes on the branch'
+              exit 1
+            fi
+            exit 0
+          ]",
```

Comment:
This checks that all local changes have been committed and merged to the main branch.

Consider whether we can (or should) also require the changes to be pushed to GitHub.